File: cli/src/main/java/com/gmail/marcosav2010/config/GeneralConfiguration.java
package com.gmail.marcosav2010.config;

import com.gmail.marcosav2010.logger.Logger;

public class GeneralConfiguration extends Configuration {

    private static final String GENERAL_CONFIG_NAME = "general";

    public GeneralConfiguration() {
        super(GENERAL_CONFIG_NAME);

        var dp = new java.util.Properties();
        Properties.propCategory.forEach((s, p) -> dp.setProperty(s, p.getDefault().toString()));

        load(dp);
    }

    @Override
    public String get(String key) {
        return get(key, Properties.propCategory.get(key).getDefault().toString());
    }

    public String getVerboseLevel() {
        return get(Logger.VERBOSE_LEVEL_PROP).toUpperCase();
    }
}

File: cli/src/main/java/com/gmail/marcosav2010/config/GeneralConfiguration.kt
package com.gmail.marcosav2010.config

import com.gmail.marcosav2010.logger.Logger

class GeneralConfiguration : Configuration(GENERAL_CONFIG_NAME) {

    companion object {
        private const val GENERAL_CONFIG_NAME = "general"
    }

    init {
        val dp = java.util.Properties()
        Properties.propCategory.forEach { (s, p) -> dp.setProperty(s, p.default.toString()) }
        load(dp)
    }

    override fun get(key: String) = get(key, Properties.propCategory[key]!!.default.toString())

    val verboseLevel: String
        get() = get(Logger.VERBOSE_LEVEL_PROP)!!.toUpperCase()
}